content/automate/ManageIQ/Service/Provisioning/StateMachines/ServiceProvision_Template.class/__methods__/update_serviceprovision_status.rb

```diff
@@ -24,15 +24,24 @@ def main
                 updated_message = update_status_message(prov, @handle.inputs['status'])
 
                 if @handle.root['ae_result'] == "error"
-                  @handle.create_notification(:level   => "error",
-                                              :subject => prov.miq_request,
-                                              :message => "Service Provision Error: #{updated_message}")
-                  @handle.log(:error, "Service Provision Error: #{updated_message}")
+                  check = "Service Provision Error: #{check_message(prov)}"
+                  message = "Service Provision Error: #{updated_message}"
+                  notified = @handle.vmdb('notification').where('options LIKE ?', "%#{check}%")
+                  if notified.count < 1
+                    @handle.create_notification(:level   => "error",
+                                                :subject => prov.miq_request,
+                                                :message => message)
+                    @handle.log(:error, message)
+                  end
                 end
               end
 
               private
 
+              def check_message(prov)
+                "Server [#{@handle.root['miq_server'].name}] Service [#{prov.destination.name}]"
```

Comment:
I think this check will probably not work when internationalization is enabled, because this would be checking English vs whatever language they have.

Reply:
Got you.